packages/core/src/logs/storage.ts

```diff
@@ -275,6 +353,72 @@ export async function getServerMetrics(
   };
 }
 
+/**
+ * Upsert server health (insert or update)
+ *
+ * This persists server health check results to SQLite,
+ * allowing queries to access live health status.
+ */
+export async function upsertServerHealth(
+  db: BunSQLiteDatabase<typeof schema>,
+  data: {
+    serverName: string;
+    health: HealthStatus;
+    lastCheck: string;
+    url: string;
+  },
+): Promise<void> {
+  const health: NewServerHealth = {
+    serverName: data.serverName,
+    health: data.health,
+    lastCheck: data.lastCheck,
+    url: data.url,
+  };
+
+  await db
+    .insert(serverHealth)
+    .values(health)
+    .onConflictDoUpdate({
+      target: serverHealth.serverName,
+      set: {
+        health: health.health,
+        lastCheck: health.lastCheck,
+        url: health.url,
+      },
+    });
+}
+
+/**
+ * Get server health by server name
+ *
+ * Returns null if health data doesn't exist in the database.
+ */
+export async function getServerHealth(
+  db: BunSQLiteDatabase<typeof schema>,
+  serverName: string,
+): Promise<{
```

Comment:
nit: could use the drizzle inferred type for selecting a server health record here right?